### PR TITLE
App Store Search

### LINKED ISSUE #5593: [CAL-401] search for apps
@Jaibles made a mockup, i think it's time to add it ❤️

> From [SyncLinear.com](https://synclinear.com) | [CAL-401](https://linear.app/calcom/issue/CAL-401/search-for-apps)

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Implements search for App Store page.

Closes #5593

 Loom Video: https://www.loom.com/share/38a3b97608a6483b98e6a93f045343ed

**Environment**: Staging(main branch) / Production

### Pending

- [x] Test on mobile
- [x] Check if we want to disable animation as it could be distracting

## Type of change

- [X] New feature (non-breaking change which adds functionality)

## How should this be tested?

Go to App Store page and use the search input at the top right.

### CODE REVIEW

**Comment by PeerRich** [comment]
as discussed, lets make the search below the nav on mobile and full width. other than that, looks great! one merge conflict

CleanShot 2022-11-27 at 22 30 59

File: apps/web/package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "@calcom/tsconfig": "*",
     "@calcom/ui": "*",
     "@daily-co/daily-js": "^0.26.0",
-    "@formkit/auto-animate": "^1.0.0-beta.1",
+    "@formkit/auto-animate": "^1.0.0-beta.5",
```
Comment (leog) [inline apps/web/package.json:41]: Needed the latest to be able to disable the animation when searching

File: apps/web/pages/apps/index.tsx [hunk 2]
```diff
@@ -1,21 +1,55 @@
...
   return (
-    <AppsLayout isPublic heading={t("app_store")} subtitle={t("app_store_description")}>
-      <AppStoreCategories categories={categories} />
-      <TrendingAppsSlider items={appStore} />
-      <AllApps apps={appStore} />
+    <AppsLayout
+      isPublic
+      heading={t("app_store")}
+      subtitle={t("app_store_description")}
+      actions={(className) => (
```
Comment (leog) [inline apps/web/pages/apps/index.tsx:43]: Needed to pass down different classes depending on the resolution to support mobile

File: apps/web/pages/apps/index.tsx [hunk 3]
```diff
@@ -1,21 +1,55 @@
...
+        <AppsSearch className={className} onChange={(e) => setSearchText(e.target.value)} />
+      )}>
+      {!searchText && (
+        <>
+          <AppStoreCategories categories={categories} />
+          <TrendingAppsSlider items={appStore} />
+        </>
+      )}
+      <AllApps apps={appStore} searchText={searchText} />
```
Comment (leog) [inline apps/web/pages/apps/index.tsx:52]: We pass down the `searchText` to the `AllApps` component to be able to style matching app names with the search text

File: apps/web/public/static/locales/en/common.json [hunk 4]
```diff
@@ -52,6 +52,7 @@
   "still_waiting_for_approval": "An event is still waiting for approval",
   "event_is_still_waiting": "Event request is still waiting: {{attendeeName}} - {{date}} - {{eventType}}",
   "no_more_results": "No more results",
+  "no_results": "No results",
```
Comment (leog) [inline apps/web/public/static/locales/en/common.json:55]: Used when no app matches the search text

File: apps/web/styles/globals.css [hunk 5]
```diff
@@ -499,4 +499,8 @@ hr {
   .h-screen {
     height: -webkit-fill-available;
   }
+}
+
+::-webkit-search-cancel-button {
```
Comment (leog) [inline apps/web/styles/globals.css:504]: Implementing our own search cancel button (aka clear input)

File: packages/ui/components/form/inputs/Input.tsx [hunk 6]
```diff
@@ -44,9 +44,8 @@ type InputFieldProps = {
   addOnLeading?: ReactNode;
   addOnSuffix?: ReactNode;
   inputIsFullWidth?: boolean;
-  /**@Default paddingX 12px  */
-  addOnPadding?: number;
```
Comment (leog) [inline packages/ui/components/form/inputs/Input.tsx:48]: Removing this property in favor of `AddOnClassname`

File: packages/ui/components/form/inputs/Input.tsx [hunk 7]
```diff
@@ -61,13 +60,15 @@ type AddonProps = {
   isFilled?: boolean;
   className?: string;
   error?: boolean;
-  paddingX?: number;
```
Comment (leog) [inline packages/ui/components/form/inputs/Input.tsx:64]: Removing property in favor of `addOnClassname`

File: packages/ui/components/form/inputs/Input.tsx [hunk 8]
```diff
@@ -344,7 +363,7 @@ export const InputFieldWithSelect = forwardRef<
       ref={ref}
       {...props}
       inputIsFullWidth={false}
-      addOnPadding={0}
+      addOnClassname="!px-0"
```
Comment (leog) [inline packages/ui/components/form/inputs/Input.tsx:366]: Converting padding setting from `addOnPadding` to `addOnClassname`

File: packages/ui/package.json [hunk 9]
```diff
@@ -15,7 +15,7 @@
   "dependencies": {
     "@calcom/lib": "*",
     "@calcom/trpc": "*",
-    "@formkit/auto-animate": "^1.0.0-beta.1",
+    "@formkit/auto-animate": "^1.0.0-beta.5",
```
Comment (leog) [inline packages/ui/package.json:18]: Needed the latest to be able to disable the animation when searching

File: packages/ui/v2/core/Shell.tsx [hunk 10]
```diff
@@ -183,6 +183,8 @@ type LayoutProps = {
   withoutMain?: boolean;
   // Gives you the option to skip HeadSEO and render your own.
   withoutSeo?: boolean;
+  // Gives the ability to include actions to the right of the heading
+  actions?: JSX.Element;
```
Comment (leog) [inline packages/ui/v2/core/Shell.tsx:187]: Adding the ability to include actions to the right of the title/subtitle section of the shell, where the search input gets added

File: packages/ui/v2/core/apps/AllApps.tsx [hunk 11]
```diff
@@ -61,11 +62,13 @@ function CategoryTab({ selectedCategory, categories }: CategoryTabProps) {
   return (
     <div className="relative mb-4 flex flex-col justify-between lg:flex-row lg:items-center">
       <h2 className="text-lg font-semibold text-gray-900 ">
-        {t("explore_apps", {
-          category:
-            (selectedCategory && selectedCategory[0].toUpperCase() + selectedCategory.slice(1)) ||
-            t("all_apps"),
-        })}
+        {searchText
```
Comment (leog) [inline packages/ui/v2/core/apps/AllApps.tsx:65]: Changing the title of the `AllApps` component to "Search" when there is a app search happening

File: packages/ui/v2/core/layouts/AppsLayout.tsx [hunk 12]
```diff
@@ -15,15 +15,17 @@ const tabs: HorizontalTabItemProps[] = [
...
+type AppsLayoutProps = {
+  children: React.ReactNode;
+  actions?: (className?: string) => JSX.Element;
+} & Omit<ComponentProps<typeof Shell>, "actions">;
+
+export default function AppsLayout({ children, actions, ...rest }: AppsLayoutProps) {
   return (
-    <Shell {...rest}>
-      <div className="flex flex-col xl:flex-row ">
+    <Shell {...rest} actions={actions?.("hidden sm:block")}>
```
Comment (leog) [inline packages/ui/v2/core/layouts/AppsLayout.tsx:25]: Shell version of the action needs to not be there for mobile

File: packages/ui/v2/core/layouts/AppsLayout.tsx [hunk 13]
```diff
@@ -15,15 +15,17 @@ const tabs: HorizontalTabItemProps[] = [
...
+      <div className="flex flex-col xl:flex-row">
         <div className="block lg:hidden">
-          <HorizontalTabs tabs={tabs} />
+          <HorizontalTabs tabs={tabs} actions={actions?.("ml-6 mr-4 mt-3 block sm:hidden")} />
```
Comment (leog) [inline packages/ui/v2/core/layouts/AppsLayout.tsx:28]: Mobile treatment styles for the search input

File: packages/ui/v2/core/navigation/tabs/HorizontalTabs.tsx [hunk 14]
```diff
@@ -5,9 +5,10 @@ export { HorizontalTabItem };
 export interface NavTabProps {
   tabs: HorizontalTabItemProps[];
   linkProps?: HorizontalTabItemProps["linkProps"];
+  actions?: JSX.Element;
 }
 
-const HorizontalTabs = function ({ tabs, linkProps, ...props }: NavTabProps) {
+const HorizontalTabs = function ({ tabs, linkProps, actions, ...props }: NavTabProps) {
```
Comment (leog) [inline packages/ui/v2/core/navigation/tabs/HorizontalTabs.tsx:11]: Being able to include actions to the tabs to include the search input on mobile

**Review by ciaranha — APPROVED** [review]
Design approved. Great work @leog 👏

**Comment by PeerRich** [comment]
awesoem! made some small change:

before

CleanShot 2022-11-28 at 17 46 49


after
CleanShot 2022-11-28 at 17 46 27

**Review by PeerRich — APPROVED** [review]

File: packages/ui/components/form/inputs/Input.tsx [hunk 15]
```diff
@@ -61,13 +60,15 @@ type AddonProps = {
   isFilled?: boolean;
   className?: string;
   error?: boolean;
-  paddingX?: number;
 };
 
-const Addon = ({ isFilled, children, className, error, paddingX = 12 }: AddonProps) => (
```
Comment (zomars) [inline packages/ui/components/form/inputs/Input.tsx:67]: Is the default padding lost?

Comment (leog) [reply packages/ui/components/form/inputs/Input.tsx:67]: Was covered with `px-3` here: https://github.com/calcom/cal.com/pull/5704/files#diff-81bcd4a4451f9e94b8bf86f92909396dd2dd9dd0598ae8ff185b7d0a29f9320cR68

**Review by pumfleet — APPROVED** [review]
splendid

**Comment by PeerRich** [comment]
E2E failing @zomars @leog

**Pull request merged**